src/Data/AdditiveGroup.hs

```diff
@@ -31,16 +35,25 @@ import Foreign.C.Types (CSChar, CInt, CShort, CLong, CLLong, CIntMax, CFloat, CD
 
 import Data.MemoTrie
 
+import qualified GHC.Generics as Gnrx
+import GHC.Generics (Generic, (:*:)(..))
+
 infixl 6 ^+^, ^-^
 
 -- | Additive group @v@.
 class AdditiveGroup v where
   -- | The zero element: identity for '(^+^)'
   zeroV :: v
+  default zeroV :: (Generic v, AdditiveGroup (Gnrx.Rep v ())) => v
```

Comment:
Good question... [the tutorial](http://hackage.haskell.org/package/base-4.9.1.0/docs/GHC-Generics.html#g:10) suggests to make an entire _class_ which is completely parameterised on a universal `p`, and then directly invoke that in the `default` method implementation. I.e.
```haskell
class AdditiveGroup' f where
  zeroV' :: f p
class AdditiveGroup v where
   zeroV :: v
   default zeroV :: (Generic v, AdditiveGroup' (Gnrx.Rep v)) => v
   zeroV = zeroV'
```
That would avoid instantiating `p` to anything particular at all.

But really, I don't see the point... basically that extra parameter is just _completely unused_ (for kind `*`, which is what – unlike in [linear](http://hackage.haskell.org/package/linear-1.20.6/docs/Linear-Vector.html) – we're dealing with here).

Yeah, a `type` around `Void` might be cleanest to use there. I'll introduce that, unless the other discussion points lead to something entirely different being preferrable.